plugins/modules/foreman_scc_product.py

```diff
@@ -0,0 +1,118 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+# (c) 2019 Manisha Singhal (ATIX AG)
+#
+# This file is part of Ansible
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+
+DOCUMENTATION = '''
+---
+module: foreman_scc_product
+short_description: Manage Foreman SccProduct
+description:
+  - "Manage Foreman SccProduct Entities"
+  - "This module requires the foreman_scc_manager plugin set up in the server"
+  - "See: U(https://github.com/ATIX-AG/foreman_scc_manager)"
+author:
+  - "Manisha Singhal (@manisha15) ATIX AG"
+options:
+  name:
+    description: Name of the scc_product
+    required: true
+    type: str
+  friendly_name:
+    description: Full name of the scc_product
+    required: false
+    type: str
+  organization:
+    description: Name of related organization
+    required: false
+    type: str
+  scc_account:
+    description: Name of the scc_account associated with scc_product
+    required: true
+    type: str
+  state:
+    description: State of the scc_product
+    default: subscribe
+    choices:
+      - subscribe
+    type: str
+extends_documentation_fragment: foreman
+'''
+
+EXAMPLES = '''
+- name: "Subscribe to scc_product"
+  foreman_scc_product:
+    name: "Product1"
+    scc_account_name: "Test1"
+    state: subscribe
+'''
+
+RETURN = ''' # '''
+
+
+from ansible.module_utils.foreman_helper import KatelloEntityAnsibleModule
+
+
+def main():
+    module = KatelloEntityAnsibleModule(
+        entity_spec=dict(
+            name=dict(required=True),
+            friendly_name=dict(),
+            scc_account=dict(required=True),
+            state=dict(default='subscribe', choices=['subscribe']),
```

Comment:
what is the point of `state` here if we only can have `subscribe` and that is the default?

Reply:
(and given it's a "state", should it be called "subscribed"?)

Reply:
Not having a `state` is what i think make an action module. So i guess we agree there?

Reply:
Yes, also we're not really using the entity spec, as we don't call ensure_entity?